File: dataset/output/OPENAI_gpt-4o/Task109_OPENAI_gpt-4o.java
import java.util.Scanner;
import java.util.regex.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter a string in Java:");
        String input = scanner.nextLine();
        Pattern pattern = Pattern.compile("^[a-zA-Z0-9]*$");
        Matcher matcher = pattern.matcher(input);
        if (matcher.matches()) {
            System.out.println("Valid Java Input");
        } else {
            System.out.println("Invalid Java Input");
        }
    }
}

File: dataset/output/OPENAI_gpt-4o/Task109_OPENAI_gpt-4o.cpp
#include <iostream>
#include <regex>

int main() {
    std::string user_input;
    std::cout << "Enter something: ";
    std::getline(std::cin, user_input);

    std::regex pattern("^[a-zA-Z0-9_]+$");
    
    if (std::regex_match(user_input, pattern)) {
        std::cout << "Valid input" << std::endl;
    } else {
        std::cout << "Invalid input" << std::endl;
    }
    
    return 0;
}